server/models.py

```diff
@@ -526,6 +526,13 @@ def enroll_from_form(cid, form):
 
     @staticmethod
     @transaction
+    def unenroll(enrollment):
+        db.session.delete(enrollment)
+        db.session.commit()
+        cache.delete_memoized(User.is_enrolled)
```

Comment:
We can specify the arguments to invalidate just one entry instead of clearing the entire cache.